ui/src/app/pages/admin/admin-institution-edit.tsx

```diff
@@ -0,0 +1,411 @@
+import {FadeBox} from 'app/components/containers';
+import {FlexColumn, FlexRow} from 'app/components/flex';
+import {SemiBoldHeader} from 'app/components/headers';
+import {TextArea} from 'app/components/inputs';
+import {BulletAlignedUnorderedList} from 'app/components/lists';
+import {Modal, ModalBody, ModalFooter, ModalTitle} from 'app/components/modals';
+import {TooltipTrigger} from 'app/components/popups';
+import {Scroll} from 'app/icons/scroll';
+import {TextInputWithLabel} from 'app/pages/login/account-creation/common';
+import {institutionApi} from 'app/services/swagger-fetch-clients';
+import colors from 'app/styles/colors';
+import {reactStyles, ReactWrapperBase, UrlParamsProps, withUrlParams} from 'app/utils';
+import {navigate} from 'app/utils/navigation';
+import {DuaType, Institution, OrganizationType} from 'generated/fetch';
+import * as fp from 'lodash/fp';
+import {Dropdown} from 'primereact/dropdown';
+import * as React from 'react';
+import * as validate from 'validate.js';
+
+const styles = reactStyles({
+  label: {
+    fontSize: '14px',
+    fontWeight: 600,
+    letterSpacing: 0,
+    lineHeight: '22px',
+    color: colors.primary,
+    marginTop: '2rem',
+    marginBottom: '0.3rem'
+  }
+});
+
+interface InstitutionEditState {
+  apiErrorMsg: string;
+  isAddInstitution: boolean;
+  institution: Institution;
+  invalidEmailAddress: boolean;
+  invalidEmailAddressMsg: string;
+  invalidEmailDomain: boolean;
+  invalidEmailDomainsMsg: string;
+  showOtherInstitution: boolean;
+  showBackButtonWarning: boolean;
+  showApiError: boolean;
+}
+
+let title = 'Add new Institution';
+let institutionToEdit;
+
+export class AdminInstitutionEditImpl extends React.Component<UrlParamsProps, InstitutionEditState> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      apiErrorMsg: '',
+      isAddInstitution: true,
+      institution: {
+        shortName: '',
+        displayName: '',
+        organizationTypeEnum: null
+      },
+      invalidEmailAddress: false,
+      invalidEmailAddressMsg: '',
+      invalidEmailDomain: false,
+      invalidEmailDomainsMsg: '',
+      showOtherInstitution: false,
+      showBackButtonWarning: false,
+      showApiError: false
+    };
+  }
+
+  async componentDidMount() {
+    // If institution short Name is passed in the URL get the institution details
+    if (this.props.urlParams.institutionId) {
+      institutionToEdit = await institutionApi().getInstitution(this.props.urlParams.institutionId);
+      title = institutionToEdit.displayName;
+      this.setState({
+        isAddInstitution: false,
+        institution: institutionToEdit,
+        showOtherInstitution: institutionToEdit.organizationTypeEnum === OrganizationType.OTHER
+      });
+    } else {
+      title = 'Add new Institution';
+      this.setState({isAddInstitution: true});
+    }
+
+  }
+
+  get institutionTypeOptions() {
+    const options = [
+      {label: 'Industry', value: OrganizationType.INDUSTRY},
```

Comment:
these should match the labels in admin-institution - let's make them use a common source.